glean_parser/data_review.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+Produce skeleton Data Review Requests.
+"""
+
+from pathlib import Path
+from typing import Sequence
+import re
+
+
+from . import parser
+from . import util
+
+
+def generate(
+    bug: str,
+    metrics_files: Sequence[Path],
+) -> int:
+    """
+    Commandline helper for Data Review Request template generation.
+
+    :param bug: pattern to match in metrics' bug_numbers lists.
+    :param metrics_files: List of Path objects to load metrics from.
+    :return: Non-zero if there were any errors.
+    """
+
+    metrics_files = util.ensure_list(metrics_files)
+
+    # Accept any value of expires.
+    parser_options = {
+        "allow_reserved": True,
+        "custom_is_expired": lambda expires: False,
+        "custom_validate_expires": lambda expires: True,
+    }
+    all_objects = parser.parse_objects(metrics_files, parser_options)
+
+    if util.report_validation_errors(all_objects):
+        return 1
+
+    # I tried [\W\Z] but it complained. So `|` it is.
+    reobj = re.compile(f"\\W{bug}\\W|\\W{bug}$")
+    durations = set()
+    metrics_table = ""
+    for category_name, metrics in all_objects.value.items():
+        for metric in metrics.values():
+            if not any([len(reobj.findall(bug)) == 1 for bug in metric.bugs]):
+                continue
+
+            metric_name = util.snake_case(metric.name)
+            category_name = util.snake_case(category_name)
+            one_line_desc = metric.description.replace("\n", " ")
+            sensitivity = ", ".join([s.name for s in metric.data_sensitivity])
+            last_bug = metric.bugs[-1]
+            metrics_table += f"{category_name}.{metric_name} | "
+            metrics_table += f"{one_line_desc} | {sensitivity} | {last_bug}\n"
+
+            durations.add(metric.expires)
+
+    if len(durations) == 1:
+        duration = next(iter(durations))
+        if duration == "never":
+            collection_duration = "This collection will be collected permanently."
+        else:
+            collection_duration = f"This collection has expiry '{duration}'"
+    else:
+        collection_duration = "Parts of this collection expire at different times: "
+        collection_duration += f"{durations}"
+
+    if "never" in durations:
+        collection_duration += "\n<TODO: identify at least one individual here> "
```

Comment:
Maybe don't use `<...>` here. If the markdown is ever viewed in a HTML rendered, this might be hidden.